File: Shuffle the Array/Leetcode_1470.py
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        return [nums[n+i//2] if i%2 else nums[i//2] for i in range(2*n)]

class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        res = [None]*(2*n)
        for i in range(2*n):
            if i%2: 
                res[i] = nums[n+i//2]
            else:
                res[i] = nums[i//2]
        return res

import itertools
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        pairs = list(zip(nums[:n], nums[n:]))
        return list(itertools.chain(*pairs))

import itertools
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        x, y = nums[:n], nums[n:] 
        pairs = list(zip(x, y))
        return list(itertools.chain(*pairs))

class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        x, y = nums[:n], nums[n:] 
        res = []
        for pair in zip(x, y):
            res += [*pair]
        return res
        
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        x, y = nums[:n], nums[n:] 
        res = []
        for i in range(n):
            res += [x[i], y[i]]
        return res

File: Shuffle the Array/Leetcode_1470.rs
impl Solution {
    pub fn shuffle(nums: Vec<i32>, n: i32) -> Vec<i32> { 
        let N = n as usize; 
        let mut res = Vec::with_capacity(2*N);
        let mut val: i32;
        for i in 0..2*N{
            val = if i%2==0 {nums[i/2]} else {x nums[N+i/2]}; 
            res.push(val);
        }
        return res;
    }
}


impl Solution {
    pub fn shuffle(nums: Vec<i32>, n: i32) -> Vec<i32> { 
        let N = n as usize; 
        let mut res = Vec::with_capacity(2*N);
        for i in 0..2*N{
            if (i%2==0){
                res.push(nums[i/2]);
            } else {
                res.push(nums[N+i/2]);
            } 
        }
        return res;
    }
}


impl Solution {
    pub fn shuffle(nums: Vec<i32>, n: i32) -> Vec<i32> { 
        let mut res = Vec::with_capacity(2*n as usize);
        for i in 0..2*n{
            if (i%2==0){
                res.push(nums[(i/2) as usize]);
            } else {
                res.push(nums[(n+i/2) as usize]);
            } 
        }
        return res;
    }
}
